src/LaurentPoly.jl

```diff
@@ -4,6 +4,16 @@
 #
 ###############################################################################
 
+###############################################################################
+#
+#   Data type and parent object methods
+#
+###############################################################################
+
+characteristic(R::LaurentPolyRing) = characteristic(base_ring(R))
+
+is_finite(R::LaurentPolyRing) = false
```

Comment:
this is wrong in the case of the base ring `residue_ring(ZZ, 1)[1]` (aka the zero ring)